api/src/repositories/region-repository.ts

```diff
@@ -0,0 +1,167 @@
+import SQL from 'sql-template-strings';
+import { z } from 'zod';
+import { getKnex } from '../database/db';
+import { ApiExecuteSQLError } from '../errors/api-error';
+import { RegionDetails } from '../services/bcgw-layer-service';
+import { BaseRepository } from './base-repository';
+
+export const IRegion = z.object({
+  region_id: z.number(),
+  region_name: z.string(),
+  org_unit: z.string(),
+  org_unit_name: z.string(),
+  feature_code: z.string(),
+  feature_name: z.string(),
+  object_id: z.number(),
+  geojson: z.any(),
+  geography: z.any()
+});
+
+export type IRegion = z.infer<typeof IRegion>;
+
+/**
+ * A repository class for accessing region data.
+ *
+ * @export
+ * @class RegionRepository
+ * @extends {BaseRepository}
+ */
+export class RegionRepository extends BaseRepository {
+  /**
+   *  Links given project to a list of given regions
+   *
+   * @param {number} projectId
+   * @param {number[]} regions
+   * @returns {*} {Promise<void>}
+   */
+  async addRegionsToProject(projectId: number, regions: number[]): Promise<void> {
+    if (regions.length < 1) {
+      return;
+    }
+
+    const sql = SQL`
+      INSERT INTO project_region (
+        project_id, 
+        region_id
+      ) VALUES `;
+
+    regions.forEach((regionId, index) => {
+      sql.append(`(${projectId}, ${regionId})`);
+
+      if (index !== regions.length - 1) {
+        sql.append(',');
+      }
+    });
+
+    sql.append(';');
+
+    try {
+      await this.connection.sql(sql);
+    } catch (error) {
+      throw new ApiExecuteSQLError('Failed to execute insert SQL for project_region', [
+        'RegionRepository->addRegionsToAProject'
```

Comment:
nit: old function name? Same with the other errors in this file. Looks like its mostly just the 'A'

Reply:
updated